src/stcal/ramp_fitting/ramp_fit_class.py

```diff
@@ -131,6 +144,7 @@ def set_dqflags(self, dqflags):
         self.flags_saturated = dqflags["SATURATED"]
         self.flags_no_gain_val = dqflags["NO_GAIN_VALUE"]
         self.flags_unreliable_slope = dqflags["UNRELIABLE_SLOPE"]
+        self.flags_chargeloss = dqflags["CHARGELOSS"]
```

Comment:
Is `CHARGELOSS` only supported by `OLS_C` and not `OLS`? If so, I think (and I tested this locally) passing in `algorithm` to this method and only setting `flags_chargeloss` if `algorithm == OLS_C` should fix the romancal tests.

Reply:
The `CHARGELOSS` flag is used when selecting either `OLS_C` or `OLS`.  Currently, the use of the `CHARGELOSS` flag is handled in the `JWST` step code.  Unfortunately, there is a bug in the segmentation computation (that occurs very rarely on an edge case) and with a computation hit.  

When running the `OLS` code, there is a message that the code returned from the `STCAL` code, but there can be a noticeable lag between that and the completion of the ramp fitting step.  To fix the bug and to improve computational performance, I moved the read noise variance re-calculation from the `JWST` step code into the C-extension.  This required some accounting changes in `ramp_fit_class.py` and `ramp_fit.py` to get the proper information to the C-extension.